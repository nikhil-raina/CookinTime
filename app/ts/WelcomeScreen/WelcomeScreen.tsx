/* eslint-disable prettier/prettier */
/**
 * @providesModule WelcomeScreen
 * @flow
 */

import React from 'react';
import styles from './WelcomeScreenStyle';
import images from '../CIT-Modules/ImageTree';

import {
  View,
  Image,
  Text,
  ImageBackground,
  TouchableOpacity,
} from 'react-native';
// import { TouchableOpacity } from 'react-native-gesture-handler';
import { ReactPropTypes } from 'react';

interface Props {
  navigation: ReactPropTypes,
}
export default class HomeScreen extends React.Component<Props, {}> {

  constructor(props: any) {
    super(props);
  }

  _navigation = () => {
    this.props.navigation.navigate('Sign In');
  }

  render(): React.ReactElement {
    const background_image = images.WELCOME_BACKGROUND;
    const logo_image = images.LOGO;
    const logo_image_shadow = images.LOGO_SHADOW;
    const appName = 'COOKIN TIME';
    const motoMessage = 'make food wait for you';
    console.log('the props',this.props.navigation);
    return (
      <TouchableOpacity onPress={this._navigation}>
        <ImageBackground  source={background_image}
                          style={styles.backgroundImage}>
          <View style={styles.container}>
              <ImageBackground  source={logo_image_shadow}
                                style={styles.logoStyleShadow}>
                <Image  source={logo_image}
                        style={styles.logoStyle}/>
              </ImageBackground>
            <Text style={styles.headerTextContainer}>
              {appName}
            </Text>
            <Text style={styles.motoTextContainer}>
              {motoMessage}
            </Text>
          </View>
        </ImageBackground>
      </TouchableOpacity>
    );
  }
}
