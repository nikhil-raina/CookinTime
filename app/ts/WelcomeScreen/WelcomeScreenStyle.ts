/* eslint-disable prettier/prettier */
/**
 * @provideModule WelcomeScreenStyle
 * @flow
 */

import {
  StyleSheet,
  Dimensions,
  Platform,
} from 'react-native';


export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },

  backgroundImage: {
    height: Dimensions.get('window').height,
    width: Dimensions.get('window').width,
    backgroundColor: 'rgba(0,0,0,0.6)',

  },

  logoContainer: {
    borderRadius: 25,
    ...Platform.select({
      ios: {
        shadowColor: '#000',
        shadowOffset: {
          width: 0,
          height: 2,
        },
        shadowOpacity: 0.8,
        shadowRadius: 5,
      },
      android: {
        elevation: 20,
      },
    }),
  },

  logoStyle: {
    height: 110,
    width: 110,
    borderRadius: 25,
  },

  headerTextContainer: {
    fontFamily: 'JosefinSans-Medium',
    fontSize: 40,
    color: '#F2E6C2',
    marginTop: 50,
  },

  motoTextContainer: {
    fontFamily: 'JosefinSans-Light',
    fontSize: 22,
    color: '#F2E6C2',
    marginBottom: 80,
  },
});
