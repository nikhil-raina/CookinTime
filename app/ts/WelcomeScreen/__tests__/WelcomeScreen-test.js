/* eslint-disable prettier/prettier */
import React from 'react';
import renderer from 'react-test-renderer';

import WelcomeScreen from '../WelcomeScreen';

describe('Rendering basic demo screen', () => {
  it('renders correctly', () => {
    const screen = renderer.create(<WelcomeScreen />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
