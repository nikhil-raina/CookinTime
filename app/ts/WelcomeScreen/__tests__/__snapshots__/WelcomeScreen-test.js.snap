// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering basic demo screen renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.6)",
        "height": 1334,
        "width": 750,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/images/WelcomeScreen/drawable-hdpi/WelcomeScreen-Background.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "alignItems": "center",
            "height": 135,
            "width": 135,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/WelcomeScreen/drawable-hdpi/WelcomeScreen-LogoShadow.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 135,
                "width": 135,
              },
              undefined,
            ]
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/WelcomeScreen/WelcomeScreen-Logo.png",
            }
          }
          style={
            Object {
              "borderRadius": 25,
              "height": 110,
              "width": 110,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#F1E5C2",
            "fontFamily": "JosefinSans-Medium",
            "fontSize": 36,
            "marginTop": 40,
          }
        }
      >
        COOKIN TIME
      </Text>
      <Text
        style={
          Object {
            "color": "#F1E5C2",
            "fontFamily": "JosefinSans-Light",
            "fontSize": 20,
            "marginBottom": 80,
          }
        }
      >
        make food wait for you
      </Text>
    </View>
  </View>
</View>
`;
